org.adempiere.base/src/org/adempiere/util/ModelClassGenerator.java

```diff
@@ -79,12 +80,15 @@ public ModelClassGenerator (int AD_Table_ID, String directory, String packageNam
 	{
 		this.packageName = packageName;
 
+		MTable table = MTable.get(AD_Table_ID);
+		boolean uuidTable = table.getKeyColumns().length != 1;

```

Comment:
don't quite understand this one, isn't uuidTable is table with 1 UUID keycolumn ?

Reply:
The model class generator works generating a lot of methods related to the ID key column.

Those methods are basically worthless in the case of multi-parent-key tables, so, my idea here is that for all tables managing multi-parent key the ModelClassGenerator consider this table as a UUID based table.

This makes two changes in the generated class:
1 - in the toString instead of calling get_ID it calls get_UUID
2 - instead of generating a method getKeyNamePair (ID+Name), it generates a method getValueNamePair (UUID+Name)

And that's all, I tried to make it different, for example not generating the ID constructors and instead generating just the UUID constructors, but that breaks a lot of M classes.

Also, I found an issue with the class MRefTable (and probably there are other cases) where the ID is AD_Reference, but is not defined as the key, but as the parent key.  That forced me to keep the X_ class as it was before (just added the UUID constructors).

Reply:
BTW - do you think is OK to generate all classes after we move these changes on Model Generator ahead?

Reply:
@hengsin - I changed the name of the variable to be more accurate about its purpose.

I generated model classes for a new test table, and also generated model classes for Test table to illustrate how would be the changes on all the model classes.